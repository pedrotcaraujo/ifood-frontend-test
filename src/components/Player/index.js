import React, { Fragment, Component } from 'react';
import './Player.css'

import PlaylistsStore from '../../stores/PlaylistsStore'

class Player extends Component {
    state = {
        playing: false
    }

    componentDidMount() {
        PlaylistsStore.addListener(this.onPlaylistsChange);
    }

    componentWillUnmount() {
        PlaylistsStore.removeListener(this.onPlaylistsChange);
    }

    onPlaylistsChange = () => {
        this.setState({
            playing: true,
            data: PlaylistsStore.getState().playing
        })
    }
    render() {
        return (
            <Fragment>
                {this.state.playing && (
                    <div className="Player">
                        <iframe title="spotify player" src={`https://open.spotify.com/embed?uri=${this.state.data.uri}`} width="100%" height="85px" frameBorder="0" allowtransparency="true"></iframe>
                    </div>
                )}
            </Fragment>
        )
    }
}

export default Player;