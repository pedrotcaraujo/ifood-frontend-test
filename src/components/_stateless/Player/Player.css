.Player {
    position: fixed;
    bottom: -10px;
    width: 100%;
    z-index: 3
}