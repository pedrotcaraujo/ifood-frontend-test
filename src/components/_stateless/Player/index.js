import React from 'react';
import PropTypes from 'prop-types'
import './Player.css'

const Player = ({ data }) => (
    <div className="Player">
        <iframe title="spotify player" src={`https://open.spotify.com/embed?uri=${data.uri}`} width="100%" height="85px" frameBorder="0" allowtransparency="true"></iframe>
    </div>
)

Player.propTypes = {
    data: PropTypes.object.isRequired
}

export default Player;