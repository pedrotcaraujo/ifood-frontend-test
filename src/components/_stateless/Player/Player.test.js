import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Player from './index';


it('renders correctly', () => {
  const tree = renderer
    .create(<Player data={{}} />)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Player data={{}} />, div);
  ReactDOM.unmountComponentAtNode(div);
});
