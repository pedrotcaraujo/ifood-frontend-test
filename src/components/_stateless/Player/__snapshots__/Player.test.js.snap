// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Player"
>
  <iframe
    allowtransparency="true"
    frameBorder="0"
    height="85px"
    src="https://open.spotify.com/embed?uri=undefined"
    title="spotify player"
    width="100%"
  />
</div>
`;
